formats/Gantt/src/Gantt.php

```diff
@@ -0,0 +1,311 @@
+		} else {
+			if ( strpos( $sortKey, 'date' ) !== false ) {
+				if ( strpos( $sortKey, 'start' ) !== false ) {
+					// sort based on startDate
+					return strcmp( $a->getEarliestStartDate(), $b->getEarliestStartDate() );
+				} else {
+					// sort based on endDate
+					return strcmp( $b->getLatestEndDate(), $a->getLatestEndDate() );
+				}
+			}
+		}
+	}
+
+	private function createNewSection( $sectionID, $sectionTitle, $startDate, $endDate, $taskID ) {
+		$ganttSection = new GanttSection();
+		//check if the id in the object is realy needed or is it enough to have it as array key
+		$ganttSection->setID( $sectionID );
+		$ganttSection->setTitle( $sectionTitle );
+		$ganttSection->setEarliestStartDate( $startDate );
+		$ganttSection->setLatestEndDate( $endDate );
+		$ganttSection->addTask( $taskID );
+
+		$this->mSections[$sectionID] = $ganttSection;
+	}
+
+	/**
+	 * Creates output for mermaidjs
+	 *
+	 * @return string
+	 */
+	public function getGanttOutput() {
+
+		$sections = $this->getSections();
+		$tasks = $this->getTasks();
+		$sortKey = $this->getSortKey();
+
+		if ( !empty( $sortKey ) ) {
+			// Order Sections
+			usort( $sections, [ $this, "sortSections" ] );
+			usort( $tasks, [ $this, "sortTasks" ] );
+
+			// reorder TaskArray of current section
+			foreach ( $sections as $section ) {
+				$orderedTasks = [];
+				foreach ( $tasks as $task ) {
+
+					// check if $section->getTasks() holds ID of current task
+					if ( in_array( $task->getID(), $section->getTasks() ) ) {
+						$sectionTasks = $section->getTasks();
+						//loop through tasks of current section
+						foreach ( $sectionTasks as $taskKey => $sectionTask ) {
+							if ( $task->getID() == $sectionTask ) {
+								array_push( $orderedTasks, $sectionTask );
+							}
+						}
+					}
+				}
+				$section->setTasks( $orderedTasks );
+			}
+		}
+
+		/*
+		 * Bring the "section" with no title to the first position.
+		 * This "section" is the one that hold tasks without any section.
+		 * If we don't display it at the beginning we have to put them into a dummy section
+		 */
+		foreach ( $sections as $key => $section ) {
+			if ( $section->getTitle() == "" ) {
+				$noSection = $section;
+				unset( $sections[$key] );
+			}
+		}
+		// push now the dummy task to the first place of the array
+		if ( isset( $noSection ) ) {
+			array_unshift( $sections, $noSection );
+		}
+
+		$title = $this->getTitle();
+		$axisFormat = $this->getAxisFormat();
+
+		$mermaidOut = "gantt\n";
+		$mermaidOut .= "dateFormat YYYY-MM-DD\n";
+		$mermaidOut .= ( !empty( $title ) ) ? "title $title\n" : "";
+		$mermaidOut .= "axisFormat $axisFormat\n";
+
+		// Output section and all related Issues
+		foreach ( $sections as $section ) {
+			if ( $section->getTitle() != "" ) {
+				$mermaidOut .= "section " . $section->getTitle() . "\n";
+			}
+
+			//loop through related section tasks
+			foreach ( $section->getTasks() as $sectionTaskValue ) {
+				foreach ( $tasks as $taskObj ) {

```

Comment:
Remove sort functionality, this can be done with inline query standard sort and order property
Line: 126-182

Commit: 6aa6b90db221170166efcb53f983a06dfe394b03